### DISCUSSION TITLE
Unable to compile with typescript when upgrading from 12 to 13

### DISCUSSION THREAD

**Discussion by Darrilla** [zwave-js/zwave-js#7503] [Request Support / Investigate Issue] [answered]
### Checklist

- [X] I have read and followed the above instructions
- [X] I have checked the [troubleshooting section](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index) and my problem is **not** described there.

- [X] I have read the [changelog](https://github.com/zwave-js/node-zwave-js/blob/master/CHANGELOG.md) and my problem was **not** mentioned there or the fix **did not work**.

### Describe the issue

What is happening?

Running tsc with  zwave-js 13.0.0 as a dependency fails with:
$ tsc                                                                                                                                                                                             [8:43:32]
```
src/index.ts:6:8 - error TS2307: Cannot find module 'zwave-js' or its corresponding type declarations.
  There are types at '/Users/dburns/scratch/zwave/node_modules/zwave-js/build/index.d.ts', but this result could not be resolved under your current 'moduleResolution' setting. Consider updating to 'node16', 'nodenext', or 'bundler'.

6 } from 'zwave-js';
         ~~~~~~~~~~


Found 1 error in src/index.ts:6
```

What did you expect to happen instead?
When using 12.13.1, it compiles without error. I expect it to be able to find zwave-js

Download my project at https://github.com/Darrilla/zwave-js-problem

Anything else we should know?

### Software versions

Driver (node-zwave-js): ...13.0.0

### Z-Wave Controller (Stick/Dongle/...)

discussions

### Device information

Manufacturer: ...
Model name: ...
Node ID: ...

### Checklist

- [X] I [made sure](https://zwave-js.github.io/node-zwave-js/#/troubleshooting/index?id=rule-1-use-the-driver-log-luke) to provide a **driver log** on level **debug**.

- [X] The log includes a re-interview of the problematic device _(if applicable)_.

- [X] The log includes the problematic interaction with the device _(if applicable)_.

- [X] I provided the node ID of the problematic device _(if applicable)_.

[a comment by an automated account is not shown]

**Comment by AlCalzone** [comment, marked as answer]
We ended up discussing this on Discord, but the reason for this is this breaking change in v12.0.0:
https://zwave-js.github.io/node-zwave-js/#/getting-started/migrating/v12?id=require-nodejs-18-switch-to-typescript39s-module-resolution-strategy-node16

I'm not sure why it kept working for so long though.
